### PR TITLE
Create rng_config from python dict

### LINKED ISSUE #533: Python comparison for rng_config
Implement comparison for RNGConfig in Python.

### LINKED ISSUE #535: RNGConfig data constructor
Make a Python constructor that takes all the necessary data as input arguments instead of the configcontent.

### PR DESCRIPTION
**Pull request by ManInFez**
Creates an rng_config from a python object and feeds the config values directly to the C object

Resolves #535 
Resolves #533

### CODE REVIEW

File: python/res/enkf/rng_config.py [hunk 1]
```diff
@@ -20,19 +20,41 @@
...
+            if ConfigKeys.RANDOM_SEED in config_dict:
+                random_seed = config_dict[ConfigKeys.RANDOM_SEED]
+            store_seed = None
+            if not random_seed and ConfigKeys.STORE_SEED in config_dict:
+                store_seed = config_dict[ConfigKeys.STORE_SEED]
+            load_seed = None
+            if not random_seed and ConfigKeys.LOAD_SEED in config_dict:
+                load_seed = config_dict[ConfigKeys.LOAD_SEED]
+
+            c_ptr = self._alloc_full(random_seed, store_seed, load_seed)
```
Comment (xjules) [inline python/res/enkf/rng_config.py:55]: So what happens if seeds are None?

File: python/tests/res/enkf/test_rng_config.py [hunk 2]
```diff
@@ -67,6 +67,40 @@ def test_load_seed(self):
...
+        config["SIMULATION"]["SEED"] = {ConfigKeys.RANDOM_SEED: random_seed,
+                                        ConfigKeys.STORE_SEED: seed_store,
+                                        ConfigKeys.LOAD_SEED: seed_load
+                                        }
+
+        with TestAreaContext("rng_config") as work_area:
+            work_area.copy_directory(case_directory)
+            rng_config = RNGConfig(config_dict=config["SIMULATION"]["SEED"])
+            res_config = ResConfig(config=config)
+            self.assertTrue(rng_config == res_config.rng_config)
```
Comment (xjules) [inline python/tests/res/enkf/test_rng_config.py]: What's the point of having it divided into 3 equality tests?

File: python/res/enkf/rng_config.py [hunk 3]
```diff
@@ -20,19 +20,41 @@
...
+        elif config_dict:
+            random_seed = None
+            if ConfigKeys.RANDOM_SEED in config_dict:
+                random_seed = config_dict[ConfigKeys.RANDOM_SEED]
+            store_seed = None
+            if not random_seed and ConfigKeys.STORE_SEED in config_dict:
+                store_seed = config_dict[ConfigKeys.STORE_SEED]
+            load_seed = None
+            if not random_seed and ConfigKeys.LOAD_SEED in config_dict:
+                load_seed = config_dict[ConfigKeys.LOAD_SEED]
```
Comment (andreabrambilla) [inline python/res/enkf/rng_config.py]: This piece of code can be rewritten with 3 one-liners such as
```Python
random_seed = config_dict.get(ConfigKeys.RANDOM_SEED, default=None)
```
The `default=None` part can actually be omitted

File: python/res/enkf/rng_config.py (hunk 1 above)
Comment (ManInFez) [reply python/res/enkf/rng_config.py:55]: Old constructor also allowed all seeds to be None. Configsuite should most likely handle this, if there should be a default value.

File: python/tests/res/enkf/test_rng_config.py [hunk 4]
```diff
@@ -67,6 +67,36 @@ def test_load_seed(self):
 
             self.assertIsNone(res_config.rng_config.random_seed)
 
+    def test_dict_constructor(self):
```
Comment (xjules) [inline python/tests/res/enkf/test_rng_config.py]: Could you rename to test_rng_dict_constructor?

**Review by xjules — APPROVED** [review]
Tested and worked! Good job!

**Pull request merged**